File: rb/features.rb
# CustomTempMail SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module CustomTempMailFeatures
  def self.make_feature(name)
    case name
    when "base"
      CustomTempMailBaseFeature.new
    when "ratelimit"
      CustomTempMailRatelimitFeature.new
    when "retry"
      CustomTempMailRetryFeature.new
    when "test"
      CustomTempMailTestFeature.new
    when "timeout"
      CustomTempMailTimeoutFeature.new
    else
      CustomTempMailBaseFeature.new
    end
  end
end

File: py/features.py
# CustomTempMail SDK feature factory

from feature.base_feature import CustomTempMailBaseFeature
from feature.test_feature import CustomTempMailTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CustomTempMailBaseFeature(),
        "test": lambda: CustomTempMailTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
